src/BenchmarksApps/SslStream/Server/Program.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Net;
+using System.Net.Security;
+using System.Security.Authentication;
+using System.Net.Sockets;
+using System.CommandLine;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Crank.EventSources;
+
+using SslStreamServer;
+
+internal class Program
+{
+    private static async Task<int> Main(string[] args)
+    {
+        var rootCommand = new RootCommand("SslStream benchmark server");
+        OptionsBinder.AddOptions(rootCommand);
+        rootCommand.SetHandler<ServerOptions>(Run, new OptionsBinder());
+        return await rootCommand.InvokeAsync(args).ConfigureAwait(false);
+    }
+
+    static async Task Run(ServerOptions options)
+    {
+        var sslOptions = CreateSslServerAuthenticationOptions(options);
+        using var sock = new Socket(SocketType.Stream, ProtocolType.Tcp);
+        sock.Bind(new IPEndPoint(IPAddress.IPv6Any, options.Port));
+        sock.Listen();
+
+        Log($"Listening on {sock.LocalEndPoint}.");
+
+        CancellationTokenSource ctrlC = new CancellationTokenSource();
+        Console.CancelKeyPress += (s, e) =>
```

Comment:
Re: 👀:

It's just I didn't understand why we need this at all, so it looked like something leftover from debugging. Now I think I see the idea behind it. I'm still not fully convinced this needs to be this way (and I'm not sure crank uses the same event to kill the server?)

I'm not 100% sure, but I believe, the way the crank will work is to wait until the client exits, and only then kill the server... So, in my view, the only active operation at this point should be listening on a socket, and I think it shouldn't need a complicated termination, so just crank killing the process will do the trick? But I might be wrong, so LMK if I'm missing something

Reply:
It probably is not necessary at this point anymore, it was useful during debugging. On the other hand, I don't think it hurts or complicates the code much.

Reply:
I don't know what is precedent in the code base. I personally would be fine with leaving debug hooks but they should probably deserve comments or documentation .... so new persons can find and use them

Reply:
After we spoke offline about reporting connection errors, I think we need to leave this in to have opportunity to actually report them.